### ISSUE TITLE
Frida 16.1.5 has a hook problem (<= 16.1.4, No Problem)

### ISSUE THREAD

**Issue by pig837** [frida/frida#2679]
Hi, I can't execute js hook script since Frida 16.1.5.

- Environment
  1) Magisk 23.0(23000)
  2) Android 11 (SM-G986N)
  3) Python Binding: frida==16.1.5, frida-tools==12.3.0

Please reply after checking.
* Note. I can't tell you the name of the app.

```javascript
Java.deoptimizeEverything()

...

const target_lib_name = 'libtest.so'

Process.findModuleByName('linker64').enumerateSymbols().forEach(sym => {
	if(sym.name.indexOf('do_dlopen') >= 0) {
		do_dlopen = sym.address
	} else if(sym.name.indexOf('call_constructor') >= 0) {
		call_ctor = sym.address
	}
})

Java.perform(function() {
	console.log(`[*] start!!!`)

        Interceptor.attach(do_dlopen, function(args) {
		        if(args[0].readUtf8String().indexOf(target_lib_name) >= 0) { // Occurred Error

...
```

```
Error: can't decode byte 0xd1 in position 0
    at <anonymous> (frida/runtime/core.js:145)
    at <anonymous> (C:\test\hook.js:35)
Error: invalid string
    at <anonymous> (frida/runtime/core.js:145)
    at <anonymous> (frida/node_modules/frida-java-bridge/lib/env.js:602)
    at <anonymous> (frida/node_modules/frida-java-bridge/lib/env.js:122)
    at value (frida/node_modules/frida-java-bridge/lib/class-factory.js:1233)
    at e (frida/node_modules/frida-java-bridge/lib/class-factory.js:639)
    at apply (native)
    at value (frida/node_modules/frida-java-bridge/lib/class-factory.js:1137)
    at e (frida/node_modules/frida-java-bridge/lib/class-factory.js:606)
    at apply (native)
    at <anonymous> (frida/node_modules/frida-java-bridge/index.js:237)
    at apply (native)
    at ne (frida/node_modules/frida-java-bridge/lib/class-factory.js:673)
    at <anonymous> (frida/node_modules/frida-java-bridge/lib/class-factory.js:651)
Process crashed: java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.res.Resources android.app.Application.getResources()' on a null object reference

***
FATAL EXCEPTION: main
Process: com.****.abcd, PID: 28707
java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.res.Resources android.app.Application.getResources()' on a null object reference
        at android.app.ActivityThread.handleConfigurationChanged(ActivityThread.java:6440)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3996)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2325)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:246)
        at android.app.ActivityThread.main(ActivityThread.java:8633)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)
***
```

**Comment by pig837** [comment]
Reproduced
* Environment
   1) KernelSU v0.7.0(11326)
   2) Linux Kernel: 4.4.302
   3) Android 13 (SM-N950N)
   4) ROM: ProjectElixir_3.12_greatlte-13.0-20230928-1415-OFFICIAL
   5) Python Binding: frida==16.1.5, frida-tools==12.3.0
       (Note) Python Binding version is regardless of this problem, but the problem is only frida-server version.

```javascript
Java.deoptimizeEverything()

var base
var do_dlopen = null
var call_ctor = null
const target_lib_name = 'libfoo.so'

Process.findModuleByName('linker64').enumerateSymbols().forEach(sym => {
	if(sym.name.indexOf('do_dlopen') >= 0) {
		do_dlopen = sym.address
	} else if(sym.name.indexOf('call_constructor') >= 0) {
		call_ctor = sym.address
	}
})

Java.perform(function() {
	console.log(`[*] start!!!`)
	
	Interceptor.attach(do_dlopen, function(args) {
		if(args[0].readUtf8String().indexOf(target_lib_name) >= 0) {
			Interceptor.attach(call_ctor, function() {
				console.log(`[++] Loaded libfoo.so`)
			})
		}
	})
})
```

[Frida 16.1.4]
```
Spawned `owasp.mstg.uncrackable3`. Resuming main thread!
[SM N950F::owasp.mstg.uncrackable3 ]-> [*] start!!!
[++] Loaded libfoo.so
[++] Loaded libfoo.so
[++] Loaded libfoo.so
[++] Loaded libfoo.so
[++] Loaded libfoo.so
[++] Loaded libfoo.so
[++] Loaded libfoo.so
[++] Loaded libfoo.so
[++] Loaded libfoo.so
[++] Loaded libfoo.so
[++] Loaded libfoo.so
[++] Loaded libfoo.so
[++] Loaded libfoo.so
[++] Loaded libfoo.so
[++] Loaded libfoo.so
[++] Loaded libfoo.so
[++] Loaded libfoo.so
[++] Loaded libfoo.so
Process terminated
```

[Frida 16.1.5]
```
Spawned `owasp.mstg.uncrackable3`. Resuming main thread!
[SM N950F::owasp.mstg.uncrackable3 ]-> [*] start!!!
Error: can't decode byte 0xd1 in position 0
    at <anonymous> (frida/runtime/core.js:145)
    at <anonymous> (C:\Tools\Script\test_owasp.mstg.uncrackable3.js:20)
Process terminated
```

**Comment by pig837** [comment]
This problem has fixed from Frida 16.1.6
